netutil/netutil_test.go

```diff
@@ -0,0 +1,116 @@
+package netutil
+
+import (
+	"net"
+	"os"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/stretchr/testify/assert"
+)
+
+// Setup required logger and example interface names and addresses
+var (
+	logger        log.Logger = log.NewLogfmtLogger(os.Stdout)
+	testIntsAddrs            = map[string]string{
+		"privNetA": "10.6.19.34/8",
+		"privNetB": "172.16.0.7/12",
+		"privNetC": "192.168.3.29/24",
+		"pubNet":   "34.120.177.193/24",
+	}
+)
+
+// A type that implements the net.Addr interface
+// Only String() is called by netutil logic
+type MockAddr struct {
```

Comment:
[nit] No need to export it.

```suggestion
type mockAddr struct {
```

Reply:
fixed in commit a0e79a212113d965ef90a75017145a0a4fa17074